build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/types/DockerImage.java

```diff
@@ -23,49 +21,71 @@
 
-    public DockerImage(String imageId, String imageTag, String targetRepo) {
+    public DockerImage(String imageId, String imageTag, String targetRepo, ArtifactoryBuildInfoClientBuilder buildInfoClientBuilder, ArtifactoryDependenciesClientBuilder dependenciesClientBuilder, String arch, String os) {
         this.imageId = imageId;
         this.imageTag = imageTag;
         this.targetRepo = targetRepo;
+        this.buildInfoClientBuilder = buildInfoClientBuilder;
+        this.dependenciesClientBuilder = dependenciesClientBuilder;
+        this.architecture = arch;
+        this.os = os;
+    }
+
+    public DockerLayers getLayers() {
+        return layers;
     }
 
     /**
      * Check if the provided manifestPath is correct.
      * Set the manifest and imagePath in case of the correct manifest.
-     *
-     * @param manifestPath
-     * @param candidateImagePath
-     * @param dependenciesClient
-     * @throws IOException
      */
-    private void checkAndSetManifestAndImagePathCandidates(String manifestPath, String candidateImagePath, ArtifactoryDependenciesClient dependenciesClient) throws IOException {
-        String candidateManifest = getManifestFromArtifactory(dependenciesClient, manifestPath);
+    private void checkAndSetManifestAndImagePathCandidates(String manifestPath, String candidateImagePath, ArtifactoryDependenciesClient dependenciesClient, boolean isRemoteRepo) throws IOException {
+        String candidateManifest = getManifestFromArtifactory(dependenciesClient, manifestPath, isRemoteRepo);
         String imageDigest = DockerUtils.getConfigDigest(candidateManifest);
         if (imageDigest.equals(imageId)) {
             manifest = candidateManifest;
             imagePath = candidateImagePath;
+            loadLayers();
         }
     }
 
-    private String getManifestFromArtifactory(ArtifactoryDependenciesClient dependenciesClient, String manifestPath) throws IOException {
+    // Searching for image's manifest in Artifactory. For remote repositories, fat-manifest can be found instead,
+    // therefore, we must compare the fat-manifest against our image architecture to find the right digest(represent our image's manifest digest).
+    // Using the correct digest from the fat-manifest, we are able to build the path toward manifest.json in Artifactory.
+    private String getManifestFromArtifactory(ArtifactoryDependenciesClient dependenciesClient, String manifestPath, boolean isRemoteRepo) throws IOException {
         HttpResponse res = null;
         try {
-            res = dependenciesClient.downloadArtifact(manifestPath);
+            res = dependenciesClient.downloadArtifact(manifestPath + "/manifest.json");
+            return IOUtils.toString(res.getEntity().getContent());
+        } catch (Exception e) {
+            if (!isRemoteRepo) {
+                throw e;
+            }
+            res = dependenciesClient.downloadArtifact(manifestPath + "/list.manifest.json");
```

Comment:
Please add a comment explaining this logic

Reply:
I rewrote the overall comment.